### PR TITLE
`Switch`: Add an interactive example

### LINKED ISSUE #103037: `Switch`:  Add an interactive example
https://master-api.flutter.dev/flutter/material/Switch-class.html is missing an example

### Solution
Add an interactive example

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/103037

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by guidezpl — CHANGES_REQUESTED** [review]

**Review by HansMuller — COMMENTED** [review]
This is a pretty elaborate example for such a simple component.

There's no harm in including a couple of switches in the example. A simple one might be what developers that want a quick copy-and-paste target are looking for. A more complex example, that demonstrates using MaterialStateProperty values would be helpful if there was some obvious motivations for the overrides (even just a comment explaining what's going on). It would also be helpful to show that a MaterialStateProperty can be used to override the color for just one state and, by returning null for the others, leave the defaults for the other states intact.

**Comment by TahaTesser** [comment]
@HansMuller 
That makes sense, I'm on it :)

File: examples/api/lib/material/switch/switch.1.dart [hunk 1]
```diff
@@ -43,8 +43,8 @@ class _SwitchExampleState extends State<SwitchExample> {
           return Colors.amber;
         }
         // Otherwise return null to set default track color
-        // for remaining states such as when the switch
-        // hovered, focused, disabled.
+        // for remaining states such as when the switch is
+        // hovered, focused or disabled.
```
Comment (guidezpl) [inline examples/api/lib/material/switch/switch.1.dart:47]: nit: keep oxford comma

File: examples/api/lib/material/switch/switch.1.dart [hunk 2]
```diff
@@ -59,8 +59,8 @@ class _SwitchExampleState extends State<SwitchExample> {
           return Colors.grey.shade400;
         }
         // Otherwise return null to set default material color
-        // for remaining states such as when the switch
-        // hovered, focused.
+        // for remaining states such as when the switch is
+        // hovered or focused.
```
Comment (guidezpl) [inline examples/api/lib/material/switch/switch.1.dart:63]: nit: keep oxford comma

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
cc: @guidezpl

**Review by guidezpl — APPROVED** [review]

**Pull request merged**